src/view/renderer.js

```diff
@@ -707,7 +707,10 @@ export default class Renderer {
 			container.appendChild( domDocument.createTextNode( '\u00A0' ) );
 		}
 
-		// Add fake container if not already added.
+		if ( container.parentElement && container.parentElement != domRoot ) {
```

Comment:
This (and below) could simply be

```js
if ( container.parentElement ) {
    container.remove();
}

domRoot.appendChild( container );
```

So simpler, but I am not sure if we want to remove and append dom element (`container`) if there's no need for it. Can it break something? If not, we can go with simpler solution.

Reply:
I'd want to avoid moving it around. That may do some strange things to focus. I even considered leaving just one line: `domRoot.appendChild( container );` (and no conditions), but I'm afraid what it may do. It depends on how it's implemented internally in the browsers. 

However, if you feel braver than I, then you can test it :D

Reply:
I may feel brave but I also feel lazy.

Reply:
AFAICS:

```js
domRoot.appendChild( container );
```

1. It works
2. Tests pass
3. Memory tests for `editor-classic` passes
4. No increase in detached DOM nodes (I was worrying that it may happen).

Tested on Chrome only though.

ps.: to be clear:
![selection_201](https://user-images.githubusercontent.com/247363/52964194-0a2dc100-33a2-11e9-9c0f-44140670d266.png)

Reply:
> Tested on Chrome only though.

Which means nothing ;) It has to be tested thoroughly in all browsers. That's why I'd choose the safe path myself.